cpp/CMakeLists.txt

```diff
@@ -50,6 +50,32 @@ set_target_properties(gtest PROPERTIES
 include_directories("${GTEST_INCLUDE_DIR}")
 add_dependencies(gtest gtest-proj)
 
+# Install OpenMP
+
+set(OPENMP_ROOT ${PROJECT_BINARY_DIR}/openmp)
+ExternalProject_Add(openmp-proj
+  PREFIX            ${OPENMP_ROOT}
+  INSTALL_DIR       ${OPENMP_ROOT}
+  GIT_REPOSITORY    https://github.com/llvm-mirror/openmp
+  GIT_TAG           master
+  CMAKE_ARGS        "-DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>"
+                    "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}"
+                    "-DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}"
+                    "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}")
+
+set(OPENMP_INCLUDE_PATH ${OPENMP_ROOT}/include)
+set(OPENMP_LIBRARY_PATH ${OPENMP_ROOT}/lib/${CMAKE_FIND_LIBRARY_PREFIXES}omp.so)
+add_library(openmp STATIC IMPORTED)
+set_target_properties(openmp PROPERTIES
+  IMPORTED_LOCATION ${OPENMP_LIBRARY_PATH})
+include_directories("${OPENMP_INCLUDE_PATH}")
+add_dependencies(openmp openmp-proj)
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fopenmp")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fopenmp")
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fopenmp")
```

Comment:
@antonio2368 Do you have an idea why this Cmake building is not linking that `libomp.so`. The build is crashing with this output: https://github.com/memgraph/mage/runs/2872757285?check_suite_focus=true

Reply:
But I don't see anything crashing, only E2E failing.

Reply:
Try looking under `Build C++` section in Actions, it silently fails and does not compile sources.